cloud-services/mbl-cloud-client/scripts/arm_update_activate.sh

```diff
@@ -195,18 +247,38 @@ ewuc_component_filename="$2"
         exit 60
     fi
 
-    # Linux always considers the sector size to be 512 bytes, no matter what
-    # the device's actual block size is. We just let dd use its default block size and
-    # ensure the seek is always a byte count.
-    # See: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/types.h?id=v4.4-rc6#n121
-    if ! dd if="$UPDATE_PAYLOAD_DIR/$ewuc_component_filename" of="$ewuc_disk_name" oflag=seek_bytes conv=fsync seek="$ewuc_offset_bytes"; then
-        printf "Writing %s to disk failed.\n" "$ewuc_component_filename"
-        exit 61
+    # Copy files to the boot partition, and the bootloader FS partition if it exists
+    if [ ! -z "$ewuc_boot_part_mnt_point"  ]; then
+        if [ ! -z "$ewuc_fs_part_mnt_point" ]; then
+            copy_dir_or_die "$UPDATE_PAYLOAD_DIR/$ewuc_component_filename" "$ewuc_fs_part_mnt_point"
+        fi
+
+        copy_dir_or_die "$UPDATE_PAYLOAD_DIR/$ewuc_component_filename" "$ewuc_boot_part_mnt_point"
+    else
+        # Write the file to raw flash.
+        #
+        # Linux always considers the sector size to be 512 bytes, no matter what
+        # the device's actual block size is. We just let dd use its default block size and
+        # ensure the seek is always a byte count.
+        # See: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/types.h?id=v4.4-rc6#n121
+        if ! dd if="$UPDATE_PAYLOAD_DIR/$ewuc_component_filename" of="$ewuc_disk_name" oflag=seek_bytes conv=fsync seek="$ewuc_offset_bytes"; then
+            # We might be on raspberry pi 3, which has some version of dd without conv and oflag options
+            # shellcheck disable=SC2003
+            # shellcheck disable=SC1001
+            if ! dd if="$UPDATE_PAYLOAD_DIR/$ewuc_component_filename" of="$ewuc_disk_name" seek="$(expr "$ewuc_offset_bytes" \/ 512)"; then
```

Comment:
Rob and I didn't really understand it either- we don't know why rpi3 is getting a version of `dd` that doesn't have these options - it's part of busybox, just like on the other platforms, and the version of busybox is the same as on all the other platforms.

There must be some configuration that's different on rpi3 but we haven't discovered what it is yet.

Reply:
Maybe because for imx-platforms we are including this config?
https://github.com/ARMmbed/meta-mbl/blob/warrior-dev/meta-freescale-mbl/recipes-core/busybox/busybox/dd_ibs_obs.cfg

Reply:
Ah, yes - that'll be it! I probably won't have time to commonize that though.

Reply:
Extra dd line removed in a8e35f0 - I moved the `dd` config file to meta-mbl-distro in https://github.com/ARMmbed/meta-mbl/commit/2e2b44674db5eb7b600be0af9561ce715c71342e